pydynamo/storage/disk/sstableiterator.py

```diff
@@ -0,0 +1,230 @@
+                                       "There is no next")
+
+    def value(self) -> str:
+        """
+        :return: the value of the key-value-time
+        set data the iterator points to
+        """
+        self.sstable_file.seek(self.cur_offset)
+        key_size_str = self.sstable_file.read(self.INT_SIZE)
+        key_size = unpack("i", key_size_str)[0]
+        self.sstable_file.read(key_size)
+        val_size_str = self.sstable_file.read(self.INT_SIZE)
+        val_size = unpack("i", val_size_str)[0]
+        value = self.sstable_file.read(val_size).decode()
+        return value
+
+    def key(self) -> str:
+        """
+        :return: the key of the key-value-time set data the iterator points to
+        """
+        self.sstable_file.seek(self.cur_offset)
+        key_size_str = self.sstable_file.read(self.INT_SIZE)
+        key_size = unpack("i", key_size_str)[0]
+        key = self.sstable_file.read(key_size).decode()
+        return key
+
+    def timestamp(self) -> str:
+        """
+        :return: the time of the key-value-time
+        set data the iterator points to
+        """
+        self.sstable_file.seek(self.cur_offset)
+        key_size_str = self.sstable_file.read(self.INT_SIZE)
+        key_size = unpack("i", key_size_str)[0]
+        self.sstable_file.read(key_size)
+        val_size_str = self.sstable_file.read(self.INT_SIZE)
+        val_size = unpack("i", val_size_str)[0]
+        self.sstable_file.read(val_size)
+        time_size_str = self.sstable_file.read(self.INT_SIZE)
+        time_size = unpack("i", time_size_str)[0]
+        time_stamp = self.sstable_file.read(time_size).decode()
+        return time_stamp
+
+    def length(self) -> int:
+        """
+        :return: the length of the key-value-time
+        set data the iterator points to
+        """
+        self.sstable_file.seek(self.cur_offset)
+        key_size_str = self.sstable_file.read(self.INT_SIZE)
+        key_size = unpack("i", key_size_str)[0]
+        self.sstable_file.read(key_size)
+        val_size_str = self.sstable_file.read(self.INT_SIZE)
+        val_size = unpack("i", val_size_str)[0]
+        self.sstable_file.read(val_size)
+        time_size_str = self.sstable_file.read(self.INT_SIZE)
+        time_size = unpack("i", time_size_str)[0]
+        self.sstable_file.read(time_size)
+        length = self.INT_SIZE + key_size \
+            + self.INT_SIZE + val_size \
+            + self.INT_SIZE + time_size
+        return length
+
+    def get_cur_offset(self) -> int:
+        """
+        :return: the offset of data the iterator currently points to
+        """
+        return self.cur_offset
+
+    def _get_range(self, key: str):
+        """
+        :param key: the key user intends to find
+        :return: the index range for key according
+        to the index table in memory
+        """
+        start = pack("i", -1)
+        end = pack("i", -1)
+        for i in self.index_table:
```

Comment:
1. please use binary search instead
2. _get_range should return a tuple of int: (start, end)